keps/sig-storage/1495-volume-populators/README.md

```diff
@@ -82,30 +95,73 @@ flexibility will allow us to experiment with approaches for data populators.
 - Enable users to create pre-populated volumes in a manner consistent with current practice
 - Enable developers to innovate with new an interesting data sources
 - Avoid changing existing workflows and breaking existing tools with new behavior
+- Ensure that users continue to get accurate error messages and feedback about volume
+  creation progress, whether using existing methods of volume creation or specifying a
+  data source associated with a volume populator.
+- Maintain existing pattern of allowing things to happen out of order. As long as there's
+  a chance things will succeed in the future, just provide a status update and retry
+  again later.
 
 ### Non-Goals
 
 - This proposal DOES NOT recommend any specific approach for data populators. The specific
   design of how a data populator should work will be handled in a separate KEP.
-- We do not propose any new object types. Specifically, no special "populator"
-  object type. Populators should function based on their own object types.
 
 ## Proposal
 
-The validation for `DataSource` field should be removed entirely, in the fullness of time.
-
-Short term, we should add a new alpha feature gate, which allows arbitrary objects to be
-specified for the `DataSource` field, with the intention to eventually make this behavior
-the standard. 
-
-Populators will work by responding to PVC objects with a data source they understand,
-and producing a PV with the expected data, such that ordinary Kubernetes workflows are
-not disrupted. In particular the PVC should be attachable to the end user's pod the
-moment it is bound, similar to a PVC created from currently supported data sources.
+Validation for the `DataSource` field will be relaxed so that any API object (except
+core API objects other than PVCs) may be specified. For objects that are not valid data
+source, rather than relying the the API server to reject them or ignore them (the prior
+behavior was to ignore them by clearing the field) allow them and provide feedback to
+API users with events.
+
+A new controller will be introduced called `data-source-validator` which will watch PVCs
+objects and generate events on objects with data sources that are not considered valid. To
+determine which data sources are valid a new CRD will be introduced valled `VolumePopulator`
+which will allow actual data populator controllers to register the API `groups`/`kinds` that
+they understand with the `data-source-validator` controller. If a PVC has a data source that
+matches the `group`/`kind` of a registered `VolumePopulator` the controller will ignore it.
```

Comment:
"the controller will ignore it" sounds negative.  Can you clarify that the controller will recognize the data source as valid.